### ISSUE TITLE
Only ASCII alpha is valid first character of tag-name

### ISSUE THREAD

**Issue by qianL93** [fb55/htmlparser2#443]
For example
`<42>text</42>`
It should be parsed into

```json
[
    {
        "data": "<42>text</42>",
        "type": "text"
    }
]
```
But htmlparser2 parses it into 

```json
[
    {
        "type": "tag",
        "name": "42",
        "attribs": {},
        "children": [
            {
                "data": "text",
                "type": "text",
                "next": null,
                "startIndex": 4,
                "prev": null,
                "parent": "[Circular ~.0]",
                "endIndex": 8
            }
        ],
        "next": null,
        "startIndex": 0,
        "prev": null,
        "parent": null,
        "endIndex": 13
    }
]
```

This is mentioned in [html.spec](https://html.spec.whatwg.org/multipage/parsing.html#tag-open-state)

**Comment by fb55** [comment]
Happy to accept PRs for this!

**Comment by Zuckjet** [comment]
as html spec described , `<42>text</42>` should be parsed as:
```
text node : '<42>text'
comment node : '42'
```
I have opened a PR for this, please help review it.

**Comment by fb55** [comment]
Fixed in #497
